llvm/lib/Target/X86/X86InstrSystem.td

```diff
@@ -212,6 +218,12 @@ def MOV16sm : I<0x8E, MRMSrcMem, (outs SEGMENT_REG:$dst), (ins i16mem:$src),
 
 let SchedRW = [WriteSystem] in {
 def SWAPGS : I<0x01, MRM_F8, (outs), (ins), "swapgs", []>, TB;
+// LKGS instructions
+let mayLoad = 1 in
+def LKGS16m : I<0x00, MRM6m, (outs), (ins i16mem:$src), "lkgs{w}\t$src",
```

Comment:
```
Execution of LKGS causes an invalid-opcode exception (#UD) if CPL > 0 or if the
logical processor is not in 64-bit mode.
```
SideEffect?

Reply:
You need the side effect b/c

> loads the base address into the IA32_KERNEL_GS_BASE MSR
> instead of the GS segment’s descriptor cache

Reply:
[0c6410e](https://github.com/llvm/llvm-project/pull/91909/commits/0c6410e012eea7629a91796d32aaf5bd35391356)